### PR TITLE
Improve api

### PR DESCRIPTION
**Pull request by pranav-bhatt**
improved API call for `cloudevents-sdk-reqwest`, replacing `response_to_event()` and `event_to_request()`

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
Can you remove from this PR the changes to cloudevents-sdk-actix-web?

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 1]
```diff
@@ -10,6 +11,24 @@ use reqwest::header::{HeaderMap, HeaderName};
 use reqwest::Response;
 use std::convert::TryFrom;
 
+/// Extention Trait for [`Response`]
+#[async_trait]
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:15]: I'm trying to wonder if we can skip using the `async_trait` crate using a generic. I'll try locally and i'll let you know

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 2]
```diff
@@ -10,6 +11,24 @@ use reqwest::header::{HeaderMap, HeaderName};
...
+pub trait ResponseExt {
+    async fn response(self) -> Result<Event>;
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:17]: I would prefer to call this method `to_event()`

File: cloudevents-sdk-reqwest/src/client_request.rs [hunk 3]
```diff
@@ -98,7 +109,14 @@ mod tests {
...
         let client = reqwest::Client::new();
-        event_to_request(input, client.post(&url))
+        /*event_to_request(input, client.post(&url))
+        .unwrap()
+        .send()
+        .await
+        .unwrap();*/
+        client
+            .post(&url)
+            .event(input)
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_request.rs:119]: That's super nice!

**Comment by pranav-bhatt** [comment]
> 
> 
> Can you remove from this PR the changes to cloudevents-sdk-actix-web?

Done

File: cloudevents-sdk-reqwest/src/client_response.rs (hunk 1 above)
Comment (pranav-bhatt) [reply cloudevents-sdk-reqwest/src/client_response.rs:15]: Alright! But from what I know, you can't define `async` functions for a Trait function without `#[async_trait]`

File: cloudevents-sdk-reqwest/src/client_response.rs (hunk 2 above)
Comment (pranav-bhatt) [reply cloudevents-sdk-reqwest/src/client_response.rs:17]: done

File: cloudevents-sdk-reqwest/src/client_request.rs (hunk 3 above)
Comment (pranav-bhatt) [reply cloudevents-sdk-reqwest/src/client_request.rs:119]: Thanks :)

File: cloudevents-sdk-actix-web/Cargo.toml [hunk 4]
```diff
@@ -13,6 +13,7 @@ readme = "README.md"
 
 [dependencies]
 cloudevents-sdk = { version = "0.1.0", path = ".." }
+async-trait = "^0.1.33"
```
Comment (slinkydeveloper) [inline cloudevents-sdk-actix-web/Cargo.toml:16]: Remove this

File: cloudevents-sdk-reqwest/src/client_request.rs [hunk 5]
```diff
@@ -7,6 +7,17 @@ use cloudevents::Event;
 use reqwest::RequestBuilder;
 use std::str::FromStr;
 
+/// Extention Trait for [`RequestBuilderExt`]
+pub trait RequestBuilderExt {
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_request.rs:11]: Can you move this and the impl too near the `event_to_request`?

File: cloudevents-sdk-reqwest/src/client_request.rs [hunk 6]
```diff
@@ -98,7 +109,14 @@ mod tests {
             .unwrap();
 
         let client = reqwest::Client::new();
-        event_to_request(input, client.post(&url))
+        /*event_to_request(input, client.post(&url))
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_request.rs:112]: Remove this comment

File: cloudevents-sdk-reqwest/src/client_request.rs [hunk 7]
```diff
@@ -132,7 +150,15 @@ mod tests {
...
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_request.rs:153]: Remove this comment

File: cloudevents-sdk-reqwest/src/client_response.rs (hunk 1 above)
Comment (slinkydeveloper) [reply cloudevents-sdk-reqwest/src/client_response.rs:15]: yes but `async` is a syntactic sugar for <T: Future<...>> fn() -> T, so in theory it should be doable

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 8]
```diff
@@ -10,6 +11,24 @@ use reqwest::header::{HeaderMap, HeaderName};
 use reqwest::Response;
 use std::convert::TryFrom;
 
+/// Extention Trait for [`Response`]
+#[async_trait]
+pub trait ResponseExt {
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:16]: Move this and the impl near `response_to_event`

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 9]
```diff
@@ -144,10 +163,17 @@ mod tests {
...
+            .send()
+            .await
+            .unwrap()
+            .to_event()
+            .await
+            .unwrap();
 
-        let resp = response_to_event(res).await.unwrap();
-        assert_eq!(expected, resp);
+        //let resp = response_to_event(res).await.unwrap();
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:175]: remove the comment

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 10]
```diff
@@ -181,10 +207,17 @@ mod tests {
...
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:219]: remove the comment

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 11]
```diff
@@ -215,9 +248,16 @@ mod tests {
...
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:260]: remove the comment

File: cloudevents-sdk-reqwest/src/client_request.rs (hunk 5 above)
Comment (pranav-bhatt) [reply cloudevents-sdk-reqwest/src/client_request.rs:11]: cool, I'll do the same for the other one too

File: cloudevents-sdk-reqwest/src/client_response.rs [hunk 12]
```diff
@@ -107,6 +108,19 @@ pub async fn response_to_event(res: Response) -> Result<Event> {
     MessageDeserializer::into_event(ResponseDeserializer::new(h, b))
 }
 
+/// Extention Trait for [`Response`]
+#[async_trait]
+pub trait ResponseExt {
+    async fn to_event(self) -> Result<Event>;
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:114]: I think it's better to name this method `into_event`. In Rust conventionally `to_*` methods takes a ref to self, while `into_*` takes ownership (like this one)

File: cloudevents-sdk-reqwest/src/client_request.rs [hunk 13]
```diff
@@ -67,6 +67,17 @@ pub fn event_to_request(event: Event, request_builder: RequestBuilder) -> Result
     BinaryDeserializer::deserialize_binary(event, RequestSerializer::new(request_builder))
 }
 
+/// Extention Trait for [`RequestBuilderExt`]
+pub trait RequestBuilderExt {
+    fn event(self, event: Event) -> Result<RequestBuilder>;
```
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_request.rs:72]: Add a little documentation here

File: cloudevents-sdk-reqwest/src/client_response.rs (hunk 12 above)
Comment (slinkydeveloper) [inline cloudevents-sdk-reqwest/src/client_response.rs:114]: Add a little documentation

**Comment by slinkydeveloper** [comment]
You need to translate the example that uses reqwest to use these new apis

**Pull request merged**